src/Cli/Microsoft.DotNet.Cli.Utils/ExponentialRetry.cs

```diff
@@ -43,16 +41,40 @@ public static IEnumerable<TimeSpan> Intervals
                     return result;
                 }
 
-                log?.WriteLine($"Operation failed. Retry count: {count}");
                 count++;
                 if (count == maxRetryCount)
                 {
-                    throw new RetryFailedException(
+                    throw new Exception(
                         $"Retry failed for {taskDescription} after {count} times with result: {result}");
                 }
             }
             throw new Exception("Timer should not be exhausted");
         }
+        
+        public static async Task<T> ExecuteWithRetryOnFailure<T>(Func<T> action,
```

Comment:
Should this just call the other overload with `_ => false` for `shouldStopRetry`?